File: CTDL1/New folder/Common7/IDE/ItemTemplatesCache/CSharp/Office SharePoint/1033/SharePoint15WcfService/Service.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.SharePoint.Client;
using Microsoft.SharePoint.Client.EventReceivers;

namespace $rootnamespace$
{
    public class $safeitemrootname$ : IRemoteEventService
    {
$if$ ("$safeitemrootname$" == "AppEventReceiver")        /// <summary>
        /// $loc_service_comment1$
        /// </summary>
        /// <param name="properties">$loc_service_comment2$</param>
        /// <returns>$loc_service_comment3$</returns>
        public SPRemoteEventResult ProcessEvent(SPRemoteEventProperties properties)
        {
            SPRemoteEventResult result = new SPRemoteEventResult();

            using (ClientContext clientContext = TokenHelper.CreateAppEventClientContext(properties, useAppWeb: false))
            {
                if (clientContext != null)
                {
                    clientContext.Load(clientContext.Web);
                    clientContext.ExecuteQuery();
                }
            }

            return result;
        }

        /// <summary>
        /// $loc_service_comment4$
        /// </summary>
        /// <param name="properties">$loc_service_comment5$</param>
        public void ProcessOneWayEvent(SPRemoteEventProperties properties)
        {
            throw new NotImplementedException();
        }
$else$        /// <summary>
        /// $loc_service_comment6$
        /// </summary>
        /// <param name="properties">$loc_service_comment7$</param>
        /// <returns>$loc_service_comment8$</returns>
        public SPRemoteEventResult ProcessEvent(SPRemoteEventProperties properties)
        {
            SPRemoteEventResult result = new SPRemoteEventResult();

            using (ClientContext clientContext = TokenHelper.CreateRemoteEventReceiverClientContext(properties))
            {
                if (clientContext != null)
                {
                    clientContext.Load(clientContext.Web);
                    clientContext.ExecuteQuery();
                }
            }

            return result;
        }

        /// <summary>
        /// $loc_service_comment9$
        /// </summary>
        /// <param name="properties">$loc_service_comment10$</param>
        public void ProcessOneWayEvent(SPRemoteEventProperties properties)
        {
            using (ClientContext clientContext = TokenHelper.CreateRemoteEventReceiverClientContext(properties))
            {
                if (clientContext != null)
                {
                    clientContext.Load(clientContext.Web);
                    clientContext.ExecuteQuery();
                }
            }
        } $endif$
    }
}

File: CTDL1/New folder/Common7/IDE/ItemTemplates/VisualBasic/Office SharePoint/1033/SharePoint16WcfService/Service.vb
Imports Microsoft.SharePoint.Client
Imports Microsoft.SharePoint.Client.EventReceivers

Public Class $safeitemrootname$
    Implements IRemoteEventService

$if$ ("$safeitemrootname$" == "AppEventReceiver")    ''' <summary>
    ''' $loc_service_comment1$
    ''' </summary>
    ''' <param name="properties">$loc_service_comment2$</param>
    ''' <returns>$loc_service_comment3$</returns>
    Public Function ProcessEvent(ByVal properties As SPRemoteEventProperties) As SPRemoteEventResult Implements IRemoteEventService.ProcessEvent
        Dim result As New SPRemoteEventResult()

        Using clientContext As ClientContext = TokenHelper.CreateAppEventClientContext(properties, useAppWeb:=False)
            If clientContext IsNot Nothing Then
                clientContext.Load(clientContext.Web)
                clientContext.ExecuteQuery()
            End If
        End Using

        Return result
    End Function

    ''' <summary>
    ''' $loc_service_comment4$
    ''' </summary>
    ''' <param name="properties">$loc_service_comment5$</param>
    Sub ProcessOneWayEvent(ByVal properties As SPRemoteEventProperties) Implements IRemoteEventService.ProcessOneWayEvent
        Throw New NotImplementedException()
    End Sub
$else$    ''' <summary>
    ''' $loc_service_comment6$
    ''' </summary>
    ''' <param name="properties">$loc_service_comment7$</param>
    ''' <returns>$loc_service_comment8$</returns>
    Public Function ProcessEvent(ByVal properties As SPRemoteEventProperties) As SPRemoteEventResult Implements IRemoteEventService.ProcessEvent
        Dim result As New SPRemoteEventResult()

        Using clientContext As ClientContext = TokenHelper.CreateRemoteEventReceiverClientContext(properties)
            If clientContext IsNot Nothing Then
                clientContext.Load(clientContext.Web)
                clientContext.ExecuteQuery()
            End If
        End Using

        Return result
    End Function

    ''' <summary>
    ''' $loc_service_comment9$
    ''' </summary>
    ''' <param name="properties">$loc_service_comment10$</param>
    Sub ProcessOneWayEvent(ByVal properties As SPRemoteEventProperties) Implements IRemoteEventService.ProcessOneWayEvent
        Using clientContext As ClientContext = TokenHelper.CreateRemoteEventReceiverClientContext(properties)
            If clientContext IsNot Nothing Then
                clientContext.Load(clientContext.Web)
                clientContext.ExecuteQuery()
            End If
        End Using
    End Sub $endif$
End Class